File: laba1.m
clear all;
clc; 
e = 0.001; 
lyambda = 0; 
lyambda_next = 1000; 
len = input('Matrix size:'); 
disp('Matrix:'); 
for i = 1:len 
    for j = 1:len 
        A(i,j) = input(''); 
    end 
end 
disp('U:'); 
for i = 1:len 
    u_0(i) = input(''); 
end 
u = u_0'; 
u_next = zeros(len); 
while abs(lyambda_next - lyambda) > e 
    lyambda = lyambda_next; 
    u_next = A*u; 
    numerator = dot(u_next,u); 
    denominator = dot(u,u); 
    lyambda_next = numerator/denominator;                                 
    u = u_next; 
    u_next = zeros(len); 
end 
disp('Lyambda_max ='); 
disp(lyambda_next); 
A = inv(A); 
lyambda = 1000;  
lyambda_next = 0; 
u = u_0'; 
u_next = zeros(len); 
while abs(lyambda_next - lyambda) > e 
    lyambda = lyambda_next; 
    u_next = A*u; 
    numerator = dot(u_next,u); 
    denominator = dot(u,u); 
    lyambda_next = numerator/denominator; 
    u = u_next; 
    u_next = zeros(len); 
end 
disp('Lyambda_min =');
disp(lyambda_next); 

File: laba1.py
import numpy as np 
from numpy import linalg as LA 

	 
e = 0.01 
lyambda = 0 
lyambda_next = 1000 
length = int(input('Matrix size: '))
A = np.zeros((length, length))
for i in range(0, length, 1): 
	for j in range(0, length, 1): 
		A[i][j] = input('') 
u = np.zeros(length) 
u_0 = np.zeros(length) 
print('U: ') 
for i in range(0, length, 1): 
	u_0[i] = input('') 
u = u_0 
u_next = np.zeros(length) 
while abs(lyambda_next - lyambda) > e: 
	lyambda = lyambda_next 
	u_next = A.dot(u) 
	numerator = u_next.dot(u) 
	denominator = u.dot(u) 
	lyambda_next = numerator / denominator                                         
	u = u_next 
	u_next = 0 
print('Lyambda_max = ', lyambda_next) 
A = LA.inv(A) 
lyambda = 0 
lyambda_next = 1000 
u = u_0 
u_next = 0 
while abs(lyambda_next - lyambda) > e: 
	lyambda = lyambda_next 
	u_next = A.dot(u) 
	numerator = u_next.dot(u) 
	denominator = u.dot(u) 
	lyambda_next = numerator / denominator 
	u = u_next 
	u_next = 0 
print('Lyambda_min = ', lyambda_next) 
